Arrays & Clusters Caraya Conversion

joec7598 opened the pull request:
Added the Caraya vi framework to unit tests that deal with array and cluster data types. These tests are under virtual folders in the project names "Arrays" and "Clusters.".

joec7598 committed:
Arrays & Clusters Caraya Conversion

Added the Caraya vi framework to unit tests that deal with array and cluster data types. These tests are under virtual folders in the project names "Arrays" and "Clusters.".

jimkring reviewed (changes requested):
Hi Joe. Looking great!

Let's also remove the old error handling.  We don't need to raise an error, explicitly, since Caraya's Assert functions will do that for us, if the assertion fails.

![image](https://github.com/vipm-io/OpenG-Variant-Data-Library/assets/381432/d5a79f3d-0cdb-4dba-9a00-14c76115d206)

joec7598 committed:
Code review changes

Renamed test_openg_array.lvlib to test_openg_variant.lvlib
Removed previous error handling form tests.
General cleanup of code.

joec7598 committed:
The rest of the tests converted to Caraya and testing error fixes

I went ahead and completed the rest of the Caraya tests under the branch.
